includes/class-wc-payments-apple-pay-registration.php

```diff
@@ -351,6 +352,54 @@ public function verify_domain_on_updated_settings( $prev_settings, $settings ) {
 		}
 	}
 
+	/**
+	 * Apple Pay supported country codes.
+	 *
+	 * @return string[]
+	 */
+	public static function supported_applepay_country_codes(): array {
+		return [
+			'ZA', 'AU', 'CN', 'HK', 'JP', 'MO', 'NZ', 'SG', 'TW', 'AM', 'AT', 'AZ', 'BY', 'BE', 'BG', 'HR', 'CY', 'CZ', // phpcs:ignore WordPress.Arrays.ArrayDeclarationSpacing.ArrayItemNoNewLine
+			'DK', 'EE', 'FO', 'FI', 'FR', 'GE', 'DE', 'GR', 'GL', 'GG', 'HU', 'IS', 'IE', 'IM', 'IT', 'KZ', 'JE', 'LV', // phpcs:ignore WordPress.Arrays.ArrayDeclarationSpacing.ArrayItemNoNewLine
+			'LI', 'LT', 'LU', 'MT', 'MD', 'MC', 'ME', 'NL', 'NO', 'PL', 'PT', 'RO', 'SM', 'RS', 'SK', 'SI', 'ES', 'SE', // phpcs:ignore WordPress.Arrays.ArrayDeclarationSpacing.ArrayItemNoNewLine
+			'CH', 'UA', 'GB', 'VA', 'AR', 'CO', 'CR', 'BR', 'MX', 'PE', 'BH', 'IL', 'PS', 'QA', 'SA', 'AE', 'CA', 'US', // phpcs:ignore WordPress.Arrays.ArrayDeclarationSpacing.ArrayItemNoNewLine
+		];
+	}
+
+	/**
+	 * Check if merchant can register domain with Apple Pay.
+	 *
+	 * @param string $account_country - Account country.
+	 *
+	 * @return bool
+	 */
+	public function can_merchant_register_domain_with_applepay( string $account_country ): bool {
+		return $this->is_account_in_supported_applepay_countries( $account_country ) && $this->has_domain_association_file_permissions();
+	}
+
+	/**
+	 * Check if domain association file has the proper permissions.
+	 *
+	 * @return bool
+	 */
+	public function has_domain_association_file_permissions(): bool {
+		$well_known_dir = untrailingslashit( ABSPATH ) . '/' . self::DOMAIN_ASSOCIATION_FILE_DIR;
+		$full_path      = $well_known_dir . '/' . self::DOMAIN_ASSOCIATION_FILE_NAME;
+
+		return is_dir( $well_known_dir ) && is_writable( $well_known_dir ) && file_exists( $full_path ) && is_readable( $full_path );
```

Comment:
This line does too many things at once, and I'm not sure if all/any of them are correct. Please take a look at [`copy_and_overwrite_domain_association_file()`](https://github.com/Automattic/woocommerce-payments/blob/fix/4712-apple-pay-incorrect-admin-notice/includes/class-wc-payments-apple-pay-registration.php#L165-L176) before proceeding with this comment:

- `copy_and_overwrite_domain_association_file()` requires neither the file nor directory to exist in order to work, but here we require both.
- Why do we check if the file `is_readable()`? We need to either check if it's writable, or verify that it already has content, while comparing the content to what we have in the gateway.

Also, during my testing, I went ahead and created an empty, and writable file. The gateway did nothing to populate it, which is not ideal either.

IMO whenever we detect that the file does not exist, we should try creating it. If that fails, we should look for an exception or an error before displaying this warning. WDYT?

Reply:
> Also, during my testing, I went ahead and created an empty, and writable file. The gateway did nothing to populate it, which is not ideal either.

That sounds like a potential bug, but not related to this PR. Mind filing a new bug-report please, so we could look into the bug in the next sprint?

> `copy_and_overwrite_domain_association_file()` requires neither the file nor directory to exist to work, but here we require both.

Good point! This condition is not matching to the function logic.
If I follow it correctly, it either:
- file exists and is writable
- well known directory exists and is writable
- parent directory of well-known one exists and writable

@RadoslavGeorgiev, @mordeth  can you verify my understanding, please?

Reply:
The logic is correct, yes. 

> That sounds like a potential bug, but not related to this PR. Mind filing a new bug-report please, so we could look into the bug in the next sprint?

As far as I'm concerned, this PR already fixes the immediate issue and can be shipped as is. Seems like we need to properly consider the whole logic around registration, so a more complete solution in a separate PR makes more sense to me.

Reply:
In my opinion, in the case of checking if the file exists to display the error message, I think the `is_hosted_domain_association_file_up_to_date()` method should be enough. It checks if the file exists and if the content is up to date. However, the check for if the directory or the parent directory exists and is writeable should be part of the `copy_and_overwrite_domain_association_file()` when the file is created or updated. What do you think?

Reply:
@RadoslavGeorgiev, @mordeth : can you create a new issue to revisit the logic around the registration file (also to cover the condition)? Thanks!

Reply:
Meanwhile, I've updated the file permission check in 0ddcf4b7027a54c61fa83b20079b1949fbd8b48b